src/cls/journal/cls_journal.cc

```diff
@@ -567,7 +567,25 @@ int journal_client_register(cls_method_context_t hctx, bufferlist *in,
     return r;
   }
 
-  cls::journal::Client client(id, data);
+  uint64_t tag_tid = 0, entry_tid = 0;
+  cls::journal::ObjectPosition first;
+  cls::journal::ObjectSetPosition object_set_position, least;
+
+  least = cls::journal::ObjectSetPosition();
+  for (std::set<cls::journal::Client>::iterator it = clients.begin();
+	 it != clients.end(); ++it) {
+    object_set_position = (*it).commit_position;
+    if (!object_set_position.object_positions.empty()) {
+      first = object_set_position.object_positions.front();
+      if (!tag_tid || (tag_tid >= first.tag_tid) || (entry_tid >= first.entry_tid)) {
```

Comment:
I believe tag_tid can be zero -- so use an optional or a bool to know if you have a valid entry.